Sources/GoogleCloud/Storage/StorageClient.swift

```diff
@@ -43,12 +43,13 @@ public final class GoogleCloudStorageClient: StorageClient {
                                   "~/.config/gcloud/application_default_credentials.json"
 
         // A token implementing OAuthRefreshable. Loaded from credentials defined above.
-        let refreshableToken = try OAuthCredentialLoader.getRefreshableToken(credentialFilePath: preferredCredentialPath, withClient: client)
+        let refreshableToken = try OAuthCredentialLoader.getRefreshableToken(credentialFilePath: preferredCredentialPath,
+                                                                             withConfig: storageConfig, andClient: client)
 
         // Set the projectId to use for this client. In order of priority:
         // - Environment Variable (PROJECT_ID)
         // - GoogleCloudProviderConfig's .project (optionally configured)
-        guard let projectId = env["PROJECT_ID"] ?? providerconfig.project ?? (refreshableToken as? OAuthServiceAccount)?.credentials.projectId else {
+        guard let projectId = env["PROJECT_ID"] ?? (refreshableToken as? OAuthServiceAccount)?.credentials.projectId ?? storageConfig.project else {
```

Comment:
The priority of which projected to get can be considered more. Do we care more about the projectID from the credentials file? or do we care more about the projectID provided by the APIConfig?